internal/service/teams/resource_teams.go

```diff
@@ -208,32 +209,31 @@ func resourceMongoDBAtlasTeamUpdate(ctx context.Context, d *schema.ResourceData,
 			return diag.FromErr(err)
 		}
 
-		_, _, err = conn.Teams.AddUsersToTeam(ctx, orgID, teamID, newUsers)
+		_, _, err = connV2.TeamsApi.AddTeamUser(ctx, orgID, teamID, &newUsers).Execute()
 		if err != nil {
 			return diag.FromErr(fmt.Errorf(errorTeamAddUsers, err))
 		}
 	}
 
-	return resourceMongoDBAtlasTeamRead(ctx, d, meta)
+	return resourceRead(ctx, d, meta)
 }
 
-func resourceMongoDBAtlasTeamDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	conn := meta.(*config.MongoDBClient).Atlas
+func resourceDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
 	ids := conversion.DecodeStateID(d.Id())
 	orgID := ids["org_id"]
 	id := ids["id"]
 
 	err := retry.RetryContext(ctx, 1*time.Hour, func() *retry.RetryError {
-		_, err := conn.Teams.RemoveTeamFromOrganization(ctx, orgID, id)
+		_, _, err := connV2.TeamsApi.DeleteTeam(ctx, orgID, id).Execute()
 		if err != nil {
-			var target *matlas.ErrorResponse
-			if errors.As(err, &target) && target.ErrorCode == "CANNOT_DELETE_TEAM_ASSIGNED_TO_PROJECT" {
-				projectID, err := getProjectIDByTeamID(ctx, conn, id)
+			if errors.Is(err, errors.New("CANNOT_DELETE_TEAM_ASSIGNED_TO_PROJECT")) {
```

Comment:
Note: v2 SDK has different way than v1  for checking errors: https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_2_error_handling.md

Reply:
Thank! Changed it